revpi-sos

```diff
@@ -40,22 +55,25 @@ Are you going to continue (N/y)?" -n 1 t_ant
 fi
 
 echo "report generating..."
-SOSFILE=$(sudo sos report -o revpi,revpi-eep,revpi-codesys --batch | grep "/tmp/sosreport" | sed -e 's/^[ \t]*//')
-echo
-echo "Your report has been generated and saved in:"
-echo "$SOSFILE"
-echo
+sudo sos report -o revpi,revpi-eep,revpi-codesys --batch --tmp-dir "$sos_tmp_dir" >/dev/null
+sos_file=$(ls "$sos_tmp_dir"/*.tar.xz)
+
+printReportFile(){
+		echo "Your sosreport has been generated and saved in:"
+		echo "$sos_file"
+}
 
 if [ "$force" != "1" ]; then
 	read -rp "To make the report have better accessibility, the ownership of the report
-can be changed to the current user.
+can be changed to the current user by inheriting ownership from the containing folder.
 Are you going to proceed(N/y)?" -n 1 t_ant
 	if [ "$t_ant" != "y" ] && [ "$t_ant" != "Y" ]; then
-		echo
+		printReportFile
 		exit 0
 	fi
 	echo
 fi
 
-sudo chown "$USER" "$SOSFILE"
-echo "The ownership of $SOSFILE has been changed to user: $USER"
+sudo chown -R --reference="$sos_tmp_dir" "$sos_file"*
```

Comment:
Whats which the `*` at the end? This is a really bad idea. If for any circumstance `$sos_file` might be empty this will run on all files in the current directory. It will also happily chmod any additional files in the directory, which might be placed there by an malicious actor.
Never run any recursive commands with elevated privileges on an asterisk in a shell script.

Reply:
yes that is a grave mistake.

the command should not run on **$sos_file** but on **$sos_tmp_dir/** which is guaranteed to be set as the --reference part would fail elsewise.